staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -75,6 +75,9 @@ type Reflector struct {
 	ShouldResync func() bool
 	// clock allows tests to manipulate time
 	clock clock.Clock
+	// forcePagination defineds whether pagination should be force for list calls
+	// and is set based on the result of the initial list call.
+	forcePagination bool
```

Comment:
@liggitt - the second commit is implementing the approach of forcing pagination based on the initial list.
Can you please take a look and see if such approach works for you? If so, I will add unit tests, but I want to get the buy-in on the approach first.

Reply:
nit: might be easier to follow named `paginatedResult`